runtime/major_gc.c

```diff
@@ -66,6 +66,9 @@ static atomic_uintnat num_domains_to_final_update_last;
 
 static atomic_uintnat terminated_domains_allocated_words;
 
+static atomic_uintnat domain_finished_global_roots;
+static caml_plat_mutex global_roots_lock = CAML_PLAT_MUTEX_INITIALIZER;
```

Comment:
I don't think we need this lock and the atomic variable to get the exclusion we need (more is always more complex and potential for bugs in future). Please can we use just the atomic variable?